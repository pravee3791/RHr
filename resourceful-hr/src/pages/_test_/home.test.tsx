import React from 'react';
import ReactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from "../home";
import { Provider } from "react-redux";
import store from "../../stores/store";


afterEach(cleanup);

it('reders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render<any, any>(<BrowserRouter>
        <Provider store={store}>
            <Home></Home>
        </Provider>
    </BrowserRouter>, div)
})

it('checks if home exists', () => {
    const HomePage = render(<BrowserRouter>
   <Provider store={store}>
            <Home></Home>
        </Provider>
    </BrowserRouter>
    );
    const home = HomePage.container.querySelector('.home');
    expect(home).toBeTruthy();
    expect(home).toBeDefined();
})

it(' Home Page matches snapshot', () => {
    const tree = renderer.create(<BrowserRouter>
     <Provider store={store}>
            <Home></Home>
        </Provider>
    </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})







