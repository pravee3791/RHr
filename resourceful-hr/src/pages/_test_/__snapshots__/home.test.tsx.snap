// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Home Page matches snapshot 1`] = `
Array [
  <div
    className="header"
  />,
  <div
    className="uploadFile"
  >
    <div
      className="uploadButton"
      onClick={[Function]}
    >
      Go to Upload Page
    </div>
  </div>,
  <div
    className="home"
  />,
]
`;
